src/space.cpp

```diff
@@ -412,6 +412,14 @@ static iarf_e do_space(chunk_t *first, chunk_t *second, int &min_sp)
 
       if (get_chunk_parent_type(second) == CT_FOR)
       {
+         if (  chunk_is_token(first, CT_SEMICOLON)
+            && options::sp_between_semi_for_empty() != IARF_IGNORE)
+         {
+            // Add or remove space between semicolons of an empty middle part of a for statement.
+            log_rule("sp_between_semi_for_empty");
+            return(options::sp_between_semi_for_empty());
+         }
```

Comment:
we need here some code to consider
`options::sp_between_semi_for_empty() == IARF_IGNORE`

Reply:
OK, I have modified the patch so that sp_before_semi_for_empty and sp_between_semi_for_empty function independently.